src/tools/perf/lib/libperf.c

```diff
@@ -1696,19 +1696,17 @@ ucx_perf_do_warmup(ucx_perf_context_t *perf, const ucx_perf_params_t *params)
         /* Stop when reaching the deadline */
         stop_status = (ucs_get_time() > deadline) ? UCS_OK : UCS_INPROGRESS;
 
+        /* Synchronize on whether to continue or stop the warmup phase */
 #if _OPENMP
-        if (params->thread_count != 1) {
 #  pragma omp barrier
 #  pragma omp single copyprivate(status)
-            /* Synchronize on whether to continue or stop the warmup phase */
-            status = ucx_perf_test_exchange_status(perf, stop_status);
-#  pragma omp barrier
-        } else
 #endif
         {
             status = ucx_perf_test_exchange_status(perf, stop_status);
         }
-
+#if _OPENMP
```

Comment:
we can remove { and } line 1704 and 1706

Reply:
right, since the pragma applies to a single statement, but for consistency of how we use pragma in other places and avoid potential bug of adding more statements, IMO {} is beter